graph-commons/src/test/scala/io/renku/http/client/GitLabClientSpec.scala

```diff
@@ -0,0 +1,191 @@
+package io.renku.http.client
+
+import cats.effect.IO
+import cats.implicits.{catsSyntaxApplicativeErrorId, toShow}
+import cats.syntax.all._
+import com.github.tomakehurst.wiremock.client.MappingBuilder
+import com.github.tomakehurst.wiremock.client.WireMock._
+import io.renku.control.Throttler
+import io.renku.generators.CommonGraphGenerators.{oauthAccessTokens, pages, personalAccessTokens}
+import io.renku.generators.Generators.Implicits._
+import io.renku.generators.Generators._
+import io.renku.graph.model.GitLabUrl
+import io.renku.http.client.RestClientError.UnauthorizedException
+import io.renku.interpreters.TestLogger
+import io.renku.logging.TestExecutionTimeRecorder
+import io.renku.metrics.GitLabApiCallRecorder
+import io.renku.stubbing.ExternalServiceStubbing
+import io.renku.testtools.IOSpec
+import org.http4s.Method.{GET, _}
+import org.http4s.Status.{NotFound, Ok, Unauthorized}
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.{Method, Request, Response, Status, Uri}
+import org.scalacheck.Gen
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.matchers.should
+import org.scalatest.wordspec.AnyWordSpec
+
+class GitLabClientSpec
+    extends AnyWordSpec
+    with IOSpec
+    with ExternalServiceStubbing
+    with should.Matchers
+    with MockFactory {
+
+  "send" should {
+
+    "fetch json from a given page" in new TestCase {
+
+      stubFor {
+        callMethod(method, s"api/v4/$path")
+          .willReturn(
+            okJson(result)
+              .withHeader("X-Next-Page", maybeNextPage.map(_.show).getOrElse(""))
+          )
+      }
+
+      client.send(method, path, endpointName)(_ => IO(result)).unsafeRunSync() shouldBe result
+    }
+
+    "fetch commits using the given personal access token" in new TestCase {
+      val personalAccessToken = personalAccessTokens.generateOne
```

Comment:
What I was thinking here is that we could generate some access token, not specifically personalAccessToken, use the `withAccessToken` as you did and simply delete the `fetch commits using the given oauth token` spec as all will be tested in this one.